backend/src/polling_station/structs.rs

```diff
@@ -73,33 +79,126 @@ impl Validate for PollingStationResults {
+                        format!("{field_name}.votes_counts"),
+                    ],
+                    code: ValidationResultCode::EqualInput,
+                });
+            }
+        }
+
+        // F.21 (recounted = false) or F.22 (recounted = true)
+        // validate that the difference for more ballots counted is correct
+        if total_voters_counts < total_votes_counts
+            && (total_votes_counts - total_voters_counts
+                != self.differences_counts.more_ballots_count)
+        {
+            validation_results.errors.push(ValidationResult {
+                fields: vec![format!("{field_name}.more_ballots_count")],
+                code: ValidationResultCode::IncorrectDifference,
+            });
+        }
+
+        // F.23 (recounted = false) or F.24 (recounted = true)
+        // validate that the difference for fewer ballots counted is correct
+        if total_voters_counts > total_votes_counts
+            && (total_voters_counts - total_votes_counts
+                != self.differences_counts.fewer_ballots_count)
+        {
+            validation_results.errors.push(ValidationResult {
+                fields: vec![format!("{field_name}.fewer_ballots_count")],
+                code: ValidationResultCode::IncorrectDifference,
+            });
+        }
+
+        // F.25 validate that only more or fewer ballots counted is filled in when there is a difference in the totals
+        if total_voters_counts != total_votes_counts
+            && (self.differences_counts.more_ballots_count != 0
+                && self.differences_counts.fewer_ballots_count != 0)
+        {
+            validation_results.errors.push(ValidationResult {
+                fields: vec![
+                    format!("{field_name}.more_ballots_count"),
+                    format!("{field_name}.fewer_ballots_count"),
+                ],
+                code: ValidationResultCode::ConflictingDifferences,
+            });
+        }
+
+        // W.31 (recounted = false) or W.32 (recounted = true)
+        // validate that no difference should be filled in when there is no difference in the totals
+        if total_voters_counts == total_votes_counts
+            && (self.differences_counts.more_ballots_count != 0
+                || self.differences_counts.fewer_ballots_count != 0)
+        {
+            if self.differences_counts.more_ballots_count != 0 {
+                validation_results.warnings.push(ValidationResult {
+                    fields: vec![format!("{field_name}.more_ballots_count")],
+                    code: ValidationResultCode::NoDifferenceExpected,
+                });
+            }
+            if self.differences_counts.fewer_ballots_count != 0 {
+                validation_results.warnings.push(ValidationResult {
+                    fields: vec![format!("{field_name}.fewer_ballots_count")],
+                    code: ValidationResultCode::NoDifferenceExpected,
+                });
```

Comment:
I can't get these to show up in the frontend. We also don't have a test for these.

The response contains `"fields":["data.more_ballots_count"]` while I'd expect `data.differences_counts.more_ballots_count`.

Reply:
Nice catch! I have fixed the field paths, please let me know if there is any other issues